File: src/test/java/pages/LoginPage.java
package test.java.pages;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;

public class LoginPage extends BasePage {

    // String locators
    private final String locatorTextUsername = "username";
    private final String locatorTextPassword = "password";
    private final String locatorButtonSignIn = "signin-button";
    private final String locatorMessageLoginError = "message";

    // Element locators
    private By byTextUsername = By.id(locatorTextUsername);
    private By byTextPassword = By.id(locatorTextPassword);
    private By byButtonSignin = By.id(locatorButtonSignIn);
    private By byMessageLoginError = By.id(locatorMessageLoginError);

    // Constructor
    public LoginPage(WebDriver webDriver) {
        super(webDriver);
    }

    // Methods
    public void enterTextUsername(String username) {
        sendKeys(locatorTextUsername, byTextUsername, username);
    }

    public void enterTextPassword(String password) {
        sendKeys(locatorTextPassword, byTextPassword, password);
    }

    public WebDriver clickButtonSignIn() {
        clickElement(locatorButtonSignIn, byButtonSignin);
        return getWebDriver();
    }

    public WebElement getMessageLoginError() {
        return getElement(locatorMessageLoginError, byMessageLoginError);
    }
}

File: pages/login-page.ts
import { Locator, Page } from "@playwright/test";

export default class LoginPage {
  public readonly page: Page;
  public readonly usernameField: Locator;
  public readonly passwordField: Locator;
  public readonly signInButton: Locator;
  public readonly wrongcredentialsMessage: Locator;
  public readonly fieldCannotBeEmptyMessage: Locator;

  constructor(page: Page) {
    this.page = page;
    this.usernameField = page.locator("#username");
    this.passwordField = page.locator("#password");
    this.signInButton = page.locator("#signin-button");
    this.wrongcredentialsMessage = page.locator("#message");
    this.fieldCannotBeEmptyMessage = page.locator("#message");
  }
}
